src/index.tsx

```diff
@@ -53,16 +57,33 @@ const ReactP5WrapperComponent: FC<P5WrapperProps> = ({
   }, [sketch]);
 
   useIsomorphicEffect(
-    () => canvasInstanceRef.current?.updateWithProps?.(props),
+    /**
+     * The `as any` cast is begrudgingly required due to a known limitation of the TypeScript compiler as demonstrated in issues:
+     *
+     * - https://github.com/microsoft/TypeScript/issues/35858
+     * - https://github.com/microsoft/TypeScript/issues/37670
+     *
+     * Potentially this will be resolved by this PR once it is eventually merged:
+     *
+     * - https://github.com/microsoft/TypeScript/pull/42382
```

Comment:
It looks like this PR was closed a few days ago due to the lack of time by the author :(

Reply:
Perhaps an opportunity to contribute then 🙈

Reply:
We will see, I don't think it's a high priority at the moment